src/System.Private.CoreLib/shared/System/IO/StreamWriter.cs

```diff
@@ -89,18 +89,26 @@ public StreamWriter(Stream stream, Encoding encoding, int bufferSize)
         {
         }
 
-        public StreamWriter(Stream stream, Encoding encoding, int bufferSize, bool leaveOpen)
+        public StreamWriter(Stream stream, Encoding encoding = null, int bufferSize = -1, bool leaveOpen = false)
             : base(null) // Ask for CurrentCulture all the time
         {
-            if (stream == null || encoding == null)
+            if (stream == null)
             {
-                throw new ArgumentNullException(stream == null ? nameof(stream) : nameof(encoding));
+                throw new ArgumentNullException(nameof(stream));
+            }
+            if (encoding == null)
+            {
+                encoding = Encoding.UTF8;
```

Comment:
Should this be `UTF8NoBOM`?

Reply:
/cc @JeremyKuhne

Reply:
Yes, it should match what the simpler constructor would do- so `UTF8NoBOM`.

Reply:
Outside of this it looks good to me.